AE advantage/limits, wording, typos (#25)
cfrg/draft-irtf-cfrg-aead-limits: draft-irtf-cfrg-aead-limits.md

| t | Size of the authentication tag (in bits) |
| l | Length of each message (in blocks)
| s | Total plaintext length in all messages (in blocks) |
| q | Number of protected messages (AEAD encryption invocations) |
| v | Number of attacker forgery attempts (failed AEAD decryption invocations) |
| p | Adversary attack probability |
| o | Offline adversary work (in number of encryption and decryption queries; multi-user setting only) |
| u | Number of users or keys (multi-user setting only) |

For each AEAD algorithm, we define the (passive) confidentiality and (active)
integrity advantage roughly as the advantage an attacker has in breaking the
corresponding classical security property for the algorithm. Moreover, we 
define the combined authenticated encryption advantage guaranteeing both

> confidentiality and integrity against an active network attacker. Specifically:

Reviewer: ```suggestion
confidentiality and integrity against an active attacker. Specifically:
```

> - Confidentiality advantage (CA): The probability of a network-passive attacker

Reviewer: ```suggestion
- Confidentiality advantage (CA): The probability of a passive attacker
```

succeeding in breaking the confidentiality properties (IND-CPA) of the AEAD scheme.
In this document, the definition of confidentiality advantage roughly is the
probability that an attacker successfully distinguishes the ciphertext outputs
of the AEAD scheme from the outputs of a random function.

> - Integrity advantage (IA): The probability of a network-active attacker succeeding

Reviewer: ```suggestion
- Integrity advantage (IA): The probability of a active attacker succeeding
```

in breaking the integrity properties (INT-CTXT) of the AEAD scheme. In this document,
the definition of integrity advantage roughly is the probability that an attacker
is able to forge a ciphertext that will be accepted as valid.

> - Authenticated Encryption advantage (AEA): The probability of a network-active

Reviewer: ```suggestion
- Authenticated Encryption advantage (AEA): The probability of a active
```

attacker succeeding in breaking the authenticated-encryption properties of the
AEAD scheme. In this document, the definition of authenticated encryption
advantage roughly is the probability that an attacker successfully distinguishes
the ciphertext outputs of the AEAD scheme from the outputs of a random function

> OR is able to forge a ciphertext that will be accepted as valid.

Reviewer: ```suggestion
or is able to forge a ciphertext that will be accepted as valid.
```

  The authenticated encryption advantage subsumes, and can be derived as the
combination of, both CA and IA:

~~~
CA <= AEA
IA <= AEA
AEA <= CA + IA
~~~

Each application requires an individual determination of limits in order to keep CA
and IA sufficiently small.  For instance, TLS aims to keep CA below 2^-60 and IA

> below 2^-57. See {{?TLS=RFC8446}}, Section 5.5.

Reviewer 2: ...in the single-user setting.

Author: Resolved in b7ddd01

# Calculating Limits

Once upper bounds on CA, IA, or AEA are determined, this document
defines a process for determining three overall operational limits:

> - Confidentiality limit (CL): The number of messages (possibly depending on the
>   number of bytes of plaintext and maybe authenticated additional data (AAD))
>   an application can encrypt before giving the adversary a confidentiality
>   advantage higher than CA.

Reviewer 2: Maybe reword:

```suggestion
- Confidentiality limit (CL): The number of messages an application can encrypt
  before giving the adversary a confidentiality advantage higher than CA.
```

Then below:

```
When limits are expressed as a number of messages, this requires assumptions about
the size of messages and any authenticated additional data (AAD).  Limits can instead
be expressed in terms of the number of plaintext bytes and maybe AAD bytes in total.
```

Reviewer: +1 to this suggestion!

Author: Resolved in b7ddd01

> - Integrity limit (IL): The number ciphertexts (possibly depending on the
>   number of bytes of ciphertext and maybe authenticated additional data (AAD))
>   an application can decrypt, either successfully or not,
>   before giving the adversary an integrity advantage higher than IA.

Reviewer: ```suggestion
- Integrity limit (IL): The number of ciphertexts an application can decrypt, 
  either successfully or not, before giving the adversary an integrity advantage 
  higher than IA.
```

Author: Resolved in b7ddd01

- Authenticated encryption limit (AEL): The combined number of messages and

>   number of ciphertext an appliation can en-/decrypt before giving the adversary

Reviewer 2: ```suggestion
  number of ciphertexts an application can encrypt or decrypt before giving the adversary
```

>   an authenticated enryption advantage higher than AEA.

Reviewer 2: ```suggestion
  an authenticated encryption advantage higher than AEA.
```
